ShipMonitor/ShipMonitor.cs

```diff
@@ -784,8 +786,15 @@ public void HandleProfile(JObject profile)
             return variables;
         }
 
-        public void writeShips()
+        public async void writeShips()
         {
+            do
+            {
+                // Make sure the dispatcher has finished prior to writing the shipyard.
+                await Task.Delay(TimeSpan.FromSeconds(1));
```

Comment:
As you say there is probably a more idiomatic way to fix this. Meanwhile this will do as a stopgap.

Right now I only have a web browser to review in, but I wonder if there are other instances of this issue in the codebase. Could you please raise an issue for me to review the codebase for this when I get back?